src/Export/Adapters/OrderNumberAdapter.php

```diff
@@ -0,0 +1,64 @@
+<?php
+
+declare(strict_types=1);
+
+namespace FINDOLOGIC\FinSearch\Export\Adapters;
+
+use FINDOLOGIC\Export\Data\Ordernumber;
+use FINDOLOGIC\FinSearch\Utils\Utils;
+use Shopware\Core\Content\Product\ProductEntity;
+
+class OrderNumberAdapter
+{
+    public function adapt(ProductEntity $product): array
+    {
+        return array_merge($this->getParentItemOrderNumber($product), $this->getChildrenProductsOrderNumbers($product));
+    }
+
+    private function getParentItemOrderNumber(ProductEntity $product): array
+    {
+        $orderNumbers = [];
+        $orderNumber = $this->getOrderNumber($product);
+
+        if (null === $orderNumber) {
+           return [];
+        }
+
+        $orderNumbers[] = $orderNumber;
+
+        return $orderNumbers;
+    }
+
+    private function getChildrenProductsOrderNumbers(ProductEntity $product): array
+    {
+        $orderNumbers = [];
+
+        foreach ($product->getChildren() as $productEntity) {
+            $orderNumber = $this->getOrderNumber($productEntity);
+            if (null === $orderNumber) {
```

Comment:
![7182_yoda](https://user-images.githubusercontent.com/20555112/161926092-31c32b33-a6e7-4280-b90d-8024b3bc6493.png): Remove you should these yoda expressions. 

Please just check through your code, there may be other occurrences.